src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxService.cs

```diff
@@ -0,0 +1,215 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.DacFx.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Hosting;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using System;
+using System.Collections.Concurrent;
+using System.Data.SqlClient;
+using System.Threading.Tasks;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx
+{
+    /// <summary>
+    /// Main class for DacFx service
+    /// </summary>
+    class DacFxService
+    {
+        private static ConnectionService connectionService = null;
+        private SqlTaskManager sqlTaskManagerInstance = null;
+        private static readonly Lazy<DacFxService> instance = new Lazy<DacFxService>(() => new DacFxService());
+        private readonly Lazy<ConcurrentDictionary<string, DacFxOperation>> operations =
+            new Lazy<ConcurrentDictionary<string, DacFxOperation>>(() => new ConcurrentDictionary<string, DacFxOperation>());
+
+        /// <summary>
+        /// Gets the singleton instance object
+        /// </summary>
+        public static DacFxService Instance
+        {
+            get { return instance.Value; }
+        }
+
+        /// <summary>
+        /// Initializes the service instance
+        /// </summary>
+        /// <param name="serviceHost"></param>
+        public void InitializeService(ServiceHost serviceHost)
+        {
+            serviceHost.SetRequestHandler(ExportRequest.Type, this.HandleExportRequest);
+            serviceHost.SetRequestHandler(ImportRequest.Type, this.HandleImportRequest);
+            serviceHost.SetRequestHandler(ExtractRequest.Type, this.HandleExtractRequest);
+            serviceHost.SetRequestHandler(DeployRequest.Type, this.HandleDeployRequest);
+        }
+
+        /// <summary>
+        /// The collection of active operations
+        /// </summary>
+        internal ConcurrentDictionary<string, DacFxOperation> ActiveOperations => operations.Value;
+
+        /// <summary>
+        /// Handles request to export a bacpac
+        /// </summary>
+        /// <returns></returns>
+        public async Task HandleExportRequest(ExportParams parameters, RequestContext<DacFxResult> requestContext)
+        {
+            try
```

Comment:
I would recommend to use Task.Run to wrap the whole thing, take a look at the other providers for reference.

Reply:
for example, profiler.

Reply:
I will look into changing this in another PR